Resources/Computing/ParallelLibraries/Srun.py

```diff
@@ -0,0 +1,106 @@
+""" srun parallel library
+
+Allow to wrap a job/pilot into a srun call that will execute n similar tasks in parallel.
+To work, srun needs to be used in conjonction with Slurm.
+"""
+import os
+import shutil
+import commands
+import stat
+
+from DIRAC.Resources.Computing.ParallelLibraries.ParallelLibrary import ParallelLibrary
+
+
+class Srun(ParallelLibrary):
+
+  def __init__(self, workingDirectory):
+    super(Srun, self).__init__('srun', workingDirectory)
+
+  def generateWrapper(self, executableFile):
+    """
+    Associate the executable with srun, to execute the same command in parallel on multiple nodes.
+    Wrap it in a new executable file
+
+    :param str executableFile: name of the executable file to run
+    :return str: name of the wrapper that runs the executable via srun
+    """
+    # As the executable will be executed by srun, it needs GRP and OTH execution permission
+    if not os.access(executableFile, 5):
+      os.chmod(executableFile, stat.S_IRWXU | stat.S_IRGRP | stat.S_IXGRP | stat.S_IROTH | stat.S_IXOTH)
+
+    # Build the script to run the executable in parallel multiple times
+    # srun is the command to execute a task multiple time in parallel
+    # -l option: add the task ID to the output
+    # -k option: do not kill the slurm job if one of the nodes is broken
+    cmd = '#!/bin/bash\n'
+    cmd += 'srun -l -k %s' % executableFile
```

Comment:
I guess `executableFile` should be quoted with `shelex.quote`/`pipes.quote`?

Reply:
I used `six.moves.shlex_quote` as `shlex.quote` seems only available in py3.3 and `pipes.quote` seems deprecated after py2.7. Is that correct?

Reply:
That's perfect. I didn't realise it was in `six.moves` and thought you'd need to do `try...except ImportError`.